src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs

```diff
@@ -0,0 +1,416 @@
+
+                if (cachedOptions == _options)
+                {
+                    return this;
+                }
+
+                return new BraceCompletionFormattingRule(_indentStyle, cachedOptions);
+            }
+
+            public override AdjustNewLinesOperation? GetAdjustNewLinesOperation(in SyntaxToken previousToken, in SyntaxToken currentToken, in NextGetAdjustNewLinesOperation nextOperation)
+            {
+                // Eg Cases -
+                // new MyObject {
+                // new List<int> {
+                // int[] arr = {
+                //           = new[] {
+                //           = new int[] {
+                if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentToken.Parent != null &&
+                (currentToken.Parent.Kind() == SyntaxKind.ObjectInitializerExpression ||
+                currentToken.Parent.Kind() == SyntaxKind.CollectionInitializerExpression ||
+                currentToken.Parent.Kind() == SyntaxKind.ArrayInitializerExpression ||
+                currentToken.Parent.Kind() == SyntaxKind.ImplicitArrayCreationExpression))
+                {
+                    if (_options.NewLinesForBracesInObjectCollectionArrayInitializers)
+                    {
+                        return CreateAdjustNewLinesOperation(1, AdjustNewLinesOption.PreserveLines);
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                }
+
+                return base.GetAdjustNewLinesOperation(in previousToken, in currentToken, in nextOperation);
+            }
+
+            public override void AddAlignTokensOperations(List<AlignTokensOperation> list, SyntaxNode node, in NextAlignTokensOperationAction nextOperation)
+            {
+                base.AddAlignTokensOperations(list, node, in nextOperation);
+                if (_indentStyle == FormattingOptions.IndentStyle.Block)
+                {
+                    var bracePair = node.GetBracePair();
+                    if (bracePair.IsValidBracePair())
+                    {
+                        AddAlignIndentationOfTokensToBaseTokenOperation(list, node, bracePair.Item1, SpecializedCollections.SingletonEnumerable(bracePair.Item2), AlignTokensOption.AlignIndentationOfTokensToFirstTokenOfBaseTokenLine);
+                    }
+                }
+            }
+
+            public override void AddSuppressOperations(List<SuppressOperation> list, SyntaxNode node, in NextSuppressOperationAction nextOperation)
+            {
+                base.AddSuppressOperations(list, node, in nextOperation);
+
+                // remove suppression rules for array and collection initializer
+                if (node.IsInitializerForArrayOrCollectionCreationExpression())
+                {
+                    // remove any suppression operation
+                    list.RemoveAll(s_predicate);
+                }
```

Comment:
needs docs.  this says what it is doing, but not why.  i have no idea how to assess this for correctness.

Reply:
Not sure to be honest either.  It looks like it's removing suppression rules that suppress wrapping, but the formatter is magic to me.  If I remove this a bunch of tests fail with the wrong indentation.  Put a comment marking as such and moving on.